addons/hr_evaluation/hr_evaluation.py

```diff
@@ -1,375 +1,465 @@
+            self.appraisal_manager = self.employee_id.appraisal_manager
+            self.appraisal_manager_ids = self.employee_id.appraisal_manager_ids
+            self.appraisal_manager_survey_id = self.employee_id.appraisal_manager_survey_id
+            self.appraisal_colleagues = self.employee_id.appraisal_colleagues
+            self.appraisal_colleagues_ids = self.employee_id.appraisal_colleagues_ids
+            self.appraisal_colleagues_survey_id = self.employee_id.appraisal_colleagues_survey_id
+            self.appraisal_self = self.employee_id.appraisal_self
+            self.appraisal_employee = self.employee_id.appraisal_employee or self.employee_id.name
+            self.appraisal_self_survey_id = self.employee_id.appraisal_self_survey_id
+            self.appraisal_subordinates = self.employee_id.appraisal_subordinates
+            self.appraisal_subordinates_ids = self.employee_id.appraisal_subordinates_ids
+            self.appraisal_subordinates_survey_id = self.employee_id.appraisal_subordinates_survey_id
+
+    @api.one
+    @api.constrains('employee_id', 'department_id', 'date_close')
+    def _check_employee_appraisal_duplication(self):
+        """ Avoid duplication"""
+        if self.employee_id and self.department_id and self.date_close:
+            date_closed = datetime.datetime.strptime(self.date_close, DTF)
+            start_datetime = time.strftime(str(date_closed.year) + '-' + str(date_closed.month) + '-01' + ' 00:00:00')
+            end_datetime = time.strftime(str(date_closed.year) + '-' + str(date_closed.month) + '-' + str(date_closed.day) + ' 23:59:59')
+            appraisal_ids = self.search([
+                ('employee_id', '=', self.employee_id.id), ('department_id', '=', self.department_id.id),
+                ('date_close', '<=', end_datetime),
+                ('date_close', '>=', start_datetime)])
+            if len(appraisal_ids) > 1:
+                raise UserError(_("You cannot create more than one appraisal for same Month & Year"))
+
+    @api.model
+    def create_message_subscribe_users(self, subscribe_users):
+        user_ids = [emp.user_id.id for emp in subscribe_users if emp.user_id]
+        if self.employee_id.user_id:
+            user_ids.append(self.employee_id.user_id.id)
+        if self.employee_id.department_id.manager_id.user_id:
+            user_ids.append(self.employee_id.department_id.manager_id.user_id.id)
+        if self.employee_id.parent_id.user_id:
+            user_ids.append(self.employee_id.parent_id.user_id.id)
+        return self.message_subscribe_users(user_ids=user_ids)
+
+    @api.model
+    def create(self, vals):
+        res = super(hr_evaluation, self.with_context(mail_create_nolog=True)).create(vals)
+        if res.appraisal_manager_ids:
+            res.create_message_subscribe_users(res.appraisal_manager_ids)
+            res.message_post(body=_("Employee Appraisal created"), subtype="mail.mt_comment", type="notification")
```

Comment:
Why do you set mail_create_nolog, to put a standard message after that ? Does not seem necessary to bypass the standard behavior in this case.